packages/grafana-ui/src/components/uPlot/config/UPlotConfigBuilder.ts

```diff
@@ -36,6 +39,13 @@ export class UPlotConfigBuilder {
     this.hooks[type]!.push(hook as any);
   }
 
+  addThresholds(options: UPlotThresholdOptions) {
+    if (!this.thresholds[options.scaleKey]) {
+      this.thresholds[options.scaleKey] = options;
+      this.addHook('draw', getThresholdsDrawHook(options));
```

Comment:
tbe `draw` hook fires last, so you'll be painting thresholds on top of everything else. not sure if this is desired behavior. you may consider an earlier hook like `drawAxes` (to paint over grid) or `drawClear` to paint under everything. or even add and option for users to choose a layer ;)

Reply:
I'm confused. This is with current impementation (`draw`):

![image](https://user-images.githubusercontent.com/2376619/116074340-e0fbd000-a691-11eb-9363-bec880791e98.png)

This is with `drawAxes`:
![image](https://user-images.githubusercontent.com/2376619/116074217-b6aa1280-a691-11eb-9cef-6c18269dfc1d.png)

And this is with `drawClear`:
![image](https://user-images.githubusercontent.com/2376619/116074291-cde90000-a691-11eb-9739-df5896414770.png)


Can't see any difference 🤔

Reply:
you have to squint, but you can see that the yellow line is faded by the threshold band in the first image. you might be able to see the difference better if you enable points and make them large. if everything is translucent, you won't see a difference cause alpha-composition is commutative.

the last one has no difference for same reason, neither the grid nor the band has alpha 1.

Reply:
Probably it's my color blindness then:)